demo/index.js

```diff
@@ -0,0 +1,78 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { createStore } from 'redux';
+import { Provider, connect } from 'react-redux';
+import { render } from 'react-dom';
+import BrandwatchReactAuth from '../';
+
+const apiService = {
+  store: null,
+};
+
+const AUTH_RECEIVE_PROFILE = 'AUTH_RECEIVE_PROFILE';
+const authReceiveProfile = (payload) => ({ type: AUTH_RECEIVE_PROFILE, payload })
+const store = createStore((state, { type, payload }) => {
+  switch(type) {
+    case AUTH_RECEIVE_PROFILE:
+      return Object.assign({}, state, {
+        profile: payload.profile,
+      });
+    default:
+      return state;
+  }
+});
+
+class DemoContent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      token: null,
+    };
+  }
+
+  componentWillMount() {
+    apiService.store.getToken({ aud: 'brandwatch.com' }).then((token) => {
+      this.setState({ token });
+    });
+  }
+
+  render() {
+    const { brandwatchAuthLogout } = this.context;
+    const { store, profile } = this.props;
+    const { token } = this.state;
+
+    return (
+      <div>
+        <p>You are authenticated!</p>
+        { token && <p>Token: { token }</p> }
+        <pre>
+          <code>
+            { JSON.stringify(profile, null, 2) }
+          </code>
+        </pre>
+        <button onClick={ () => brandwatchAuthLogout() }>Logout</button>
+      </div>
+    );
+  }
+}
+
+DemoContent.contextTypes = {
+  brandwatchAuthLogout: PropTypes.func.isRequired,
+};
+
+const DemoContentContainer = connect((state) => ({
+  profile: state.profile,
+}), {})(DemoContent);
+
+render((
+  <BrandwatchReactAuth
+      audience="brandwatch.com"
+      domain={ __BW_REACT_AUTH_DOMAIN__ }
+      logoutRedirectLocation="http://localhost:4000"
+      onAuthenticate={ (payload) => store.dispatch(authReceiveProfile(payload)) }
+      onCreateStore={ (store) => apiService.store = store }>
+    <Provider store={ store }>
+      <DemoContentContainer />
```

Comment:
Would love to play around with a dynamic loading example at some stage, so we don't have to download the whole app before getting logged out.